### PR TITLE
Add user and group paperless and avoid to use hardcoded userId and groupId

### PR DESCRIPTION
**Pull request by michalharakal**
1000 hardwired its not working e.g. on synology.

Related to #306

### CODE REVIEW

**Comment by Tooa** [comment]
@michalharakal Is this related to https://github.com/the-paperless-project/paperless/pull/599?

File: Dockerfile [hunk 1]
```diff
@@ -48,8 +49,6 @@ RUN apk add --no-cache \
 # Create the consumption directory
     mkdir -p $PAPERLESS_CONSUMPTION_DIR && \
 # Create user
-    addgroup -g 1000 paperless && \
```
Comment (bauerj) [inline Dockerfile:51]: If you leave the comment here, you should move the command back down.

**Comment by bauerj** [comment]
What exactly is the advantage of this? Wouldn't `adduser -S paperless -G paperless` simply use the first unused UID for the new user? Since there are no other users in that Docker image that would be 1000, wouldn't it?

File: Dockerfile (hunk 1 above)
Comment (michalharakal) [reply Dockerfile:51]: Yes, it makes sense. Should be on the right place now.

**Comment by michalharakal** [comment]
I can't explain why 1000 is not working on synology. As I understood, its generally a good practice even in container not to run as root...

**Comment by michalharakal** [comment]
@Tooa yes indeed, #599 seems to be a different solution to the same problem.

**Comment by bauerj** [comment]
> As I understood, its generally a good practice even in container not to run as root.

Surely, but UID 1000 is not root (that would be UID 0). Of course I'm not opposed to any other number I just want to understand the issue in more detail.

**Comment by michalharakal** [comment]
@bauerj Meanwhile I have checked #599 and paperless starts on my synolgy out of the box, so its working for me now and my merge ist not required anymore.  I am no expert neither in linux nor in docker, but from by experience it feels somehow not realy right to depend on hard wired constants. What I already saw it that user/group and rights are handled in entrypoint script dynamically. There is a reason for that I don't know (probably issues with mounting volumens etc.)
```
#!/bin/bash

if [ -z "${USER_UID}" -o -z "${USER_GID}" ]; then
    echo "Getting UID and GID from current directory"
    CURRENT_DIR=$(ls -lnd .)
    USER_UID=$(echo "${CURRENT_DIR}" | cut -f3 -d' ')
    USER_GID=$(echo "${CURRENT_DIR}" | cut -f4 -d' ')
fi

if [ "${USER_UID}" -ne 0 ]; then
    groupadd \
        --gid ${USER_GID} \
        paperless
    
    useradd \
        --uid ${USER_UID} \
        --gid ${USER_GID} \
        --shell /bin/bash \
        paperless
fi

echo "Using UID ${USER_UID} and GID ${USER_GID} for execution"

chown --quiet --recursive "${USER_UID}:${USER_GID}" /home/paperless

echo "Executing $@"

if [ "${USER_UID}" -ne 0 ]; then
    sudo --user paperless --preserve-env --set-home -- "$@"
else
    exec "$@"
fi

```

**Comment by michalharakal** [comment]
Anyway, I can close the MR.

**Comment by bauerj** [comment]
I'm glad it works for you now! I agree that it's not a good idea to use some "magic" constants like 1000. In this case, it might be better to make that configurable.

**Pull request closed without merging**